### PR TITLE
feat: All components support isolation

### PR DESCRIPTION
**Pull request by zhenlei520**
# Description

All modules support isolation

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/DefaultIsolationConfigProvider.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+namespace Masa.BuildingBlocks.Isolation;
+
+public class DefaultIsolationConfigProvider : IIsolationConfigProvider
+{
+    private readonly IServiceProvider _serviceProvider;
+    private readonly IMultiEnvironmentContext? _environmentContext;
+    private readonly IMultiTenantContext? _tenantContext;
+    private readonly ILogger<DefaultIsolationConfigProvider>? _logger;
+    private readonly List<ModuleConfigRelationInfo> _data;
```
Comment (doddgu) [inline src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/DefaultIsolationConfigProvider.cs:13]: Module?

File: src/BuildingBlocks/Service/Masa.BuildingBlocks.Service.Caller/AbstractCaller.cs [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 namespace Masa.BuildingBlocks.Service.Caller;
 
-public abstract class AbstractCaller : ICaller
+public abstract class AbstractCaller : ICallerDisposeManual
```
Comment (doddgu) [inline src/BuildingBlocks/Service/Masa.BuildingBlocks.Service.Caller/AbstractCaller.cs:6]: IFullCaller? IManualCaller?

File: src/Contrib/Storage/ObjectStorage/Masa.Contrib.Storage.ObjectStorage.Aliyun/Internal/AliyunStorageConfigureOptions.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace Masa.Contrib.Storage.ObjectStorage.Aliyun;
+
+internal class AliyunStorageConfigureOptions : AliyunOptions
+{
+    public AliyunStorageOptions Storage { get; set; } = new();
+
+    /// <summary>
+    /// Aliyun STS configuration, it is not necessarily a storage configuration
+    ///
```
Comment (doddgu) [inline src/Contrib/Storage/ObjectStorage/Masa.Contrib.Storage.ObjectStorage.Aliyun/Internal/AliyunStorageConfigureOptions.cs:14]: ?

File: temp/WebApplication1/ApplicationServices/RegisterUser/RegisterUserEvent.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
```
Comment (doddgu) [inline temp/WebApplication1/ApplicationServices/RegisterUser/RegisterUserEvent.cs:1]: ?

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/IIsolationConfigProvider.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+public interface IIsolationConfigProvider
+{
+    /// <summary>
+    /// Get the configuration of the specified SectionName and the specified Name under the current environment or tenant
+    /// </summary>
+    /// <param name="sectionName"></param>
+    /// <param name="name"></param>
+    /// <typeparam name="TModuleConfig"></typeparam>
+    /// <returns></returns>
+    TModuleConfig? GetModuleConfig<TModuleConfig>(string sectionName, string name = "") where TModuleConfig : class;
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/IIsolationConfigProvider.cs:18]: ComponentConfig？

File: src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/Options/IsolationOptions.cs [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
-    public string ConnectionString { get; set; }
+    public bool Enable => !string.IsNullOrWhiteSpace(SectionName);
 
-    /// <summary>
-    /// Used to control the configuration with the highest score when multiple configurations are satisfied. The default score is 100
-    /// </summary>
-    public int Score { get; set; } = 100;
+    public IsolationOptions()
+    {
+        MultiTenantType = typeof(Guid);
```
[the remark this answers is not in the document]
Comment (zhenlei520) [reply src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/Options/IsolationOptions.cs:18]: MultiTenantIdType

When the multi-tenant id type is not set, Guid is used by default

File: src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/DefaultIsolationConfigProvider.cs [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+            moduleInfo = modules.First();
+        }
+        else
+        {
+            moduleInfo = null;
+        }
+        _data.Add(new ComponentConfigRelationInfo()
+        {
+            Data = moduleInfo,
+            ModuleType = typeof(TComponentConfig),
```
Comment (doddgu) [inline src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/DefaultIsolationConfigProvider.cs:80]: ModuleType?

File: src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/DefaultIsolationConfigProvider.cs [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+        }
+
+        var data = GetIsolationConfigurationOptions<TComponentConfig>(name, sectionName);
+        TComponentConfig? moduleInfo = null;
+        var modules = data
+            .Where(condition.Compile())
+            .OrderByDescending(option => option.Score)
+            .Select(option => option.Data)
+            .ToList();
+        if (modules.Count >= 1)
```
Comment (doddgu) [inline src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/DefaultIsolationConfigProvider.cs:66]: modules?

File: src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/DefaultIsolationConfigProvider.cs [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+        {
+            moduleInfo = null;
+        }
+        _data.Add(new ComponentConfigRelationInfo()
+        {
+            Data = moduleInfo,
+            ModuleType = typeof(TComponentConfig),
+            SectionName = sectionName
+        });
+        return moduleInfo;
```
Comment (doddgu) [inline src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/DefaultIsolationConfigProvider.cs:83]: moduleInfo?

File: src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/Internal/ComponentConfigRelationInfo.cs [hunk 10]
```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+// ReSharper disable once CheckNamespace
+
+namespace Masa.BuildingBlocks.Data;
+
+internal class ComponentConfigRelationInfo
+{
+    public Type ModuleType { get; set; }
```
Comment (doddgu) [inline src/BuildingBlocks/Isolation/Masa.BuildingBlocks.Isolation/Internal/ComponentConfigRelationInfo.cs:10]: Module?

**Review by doddgu — CHANGES_REQUESTED** [review]

**Review by doddgu — APPROVED** [review]

**Pull request merged**